src/hyperlink/_url.py

```diff
@@ -528,24 +577,19 @@ def _percent_decode(text, normalize_case=False, subencoding='utf-8',
     u'abc def'
 
     Args:
-       text (unicode): Text with percent-encoding present.
-       normalize_case (bool): Whether undecoded percent segments, such
-          as encoded delimiters, should be uppercased, per RFC 3986
-          Section 2.1. See :func:`_decode_path_part` for an example.
-       subencoding (unicode): The name of the encoding underlying the
-          percent-encoding. Pass `False` to get back raw bytes.
-       raise_subencoding_exc (bool): Whether an error in decoding the bytes
-          underlying the percent-decoding should be raised.
+        text: Text with percent-encoding present.
+        normalize_case: Whether undecoded percent segments, such as encoded
+            delimiters, should be uppercased, per RFC 3986 Section 2.1.
+            See :func:`_decode_path_part` for an example.
+        subencoding: The name of the encoding underlying the percent-encoding.
+        raise_subencoding_exc: Whether an error in decoding the bytes
+            underlying the percent-decoding should be raised.
 
     Returns:
-       unicode: The percent-decoded version of *text*, decoded by
-         *subencoding*, unless `subencoding=False` which returns bytes.
-
+        The percent-decoded version of *text*, decoded by *subencoding*.
     """
     try:
-        quoted_bytes = text.encode(
-            'utf-8' if subencoding is False else subencoding
-        )
+        quoted_bytes = text.encode(subencoding)
```

Comment:
Removed code path for `subencoding=`False`.
It was not used by any callers, and it required that we return a `Union[str, bytes]` where returning `Text` all the time avoids casting elsewhere.

Reply:
Yeah, subencoding is in the TODO as this... thing. So when bytes are percent encoded and stuck in the URL, theoretically they can be Latin-1 or another encoding.  Does that make sense?

My thought was that maybe the subencoding arg could bubble up to the URL API, as an escape valve for non-UTF8 subencodings.. If you have thoughts about that, I'd love to hear them. Maybe it's just adding a warning box somewhere in the hyperlink docs: "hyperlink only supports utf-8 encoded text within URL parameters, etc."

Reply:
OK that makes sense.  I'll assert that we cross that bridge when we get to it, but that it would be better to refactor this code when we do so, such that we aren't having `_percent_decode` with a polymorphic signature.

Reply:
So the challenge here is that `_percent_decode` with `subencoding=False` returns `bytes` instead of `Text` (aka `unicode`).  That would require that we annotate it as `Union[Text, bytes]` and a large number of callers would then have to `cast` the return to `Text` (or `bytes` in the future).

Even in cases where you could previously specify `subencoding=False`, there are cases in this function where you get `Text` instead of `bytes`, which seems problematic.

So what I'd suggest is that when we want to get raw bytes out, that we add a new function that always returns bytes, possibly factoring out the middle bit (for for loop) form here.

Reply:
If my understanding is correct here we could do this with an `@override`?

Reply:
Perhaps, but wouldn't it be safer to do the separate function? It's internal API, so type checking here is for us to catch our own mistakes.

Reply:
Yeah, I'm fine handling this as a refactor later. We'll always have the git history :)